Lots of components

NerdDiffer opened the pull request:
### To get this to run on Android:

edit the file: 'client/zestyreactnative/src/helpers/scaffolding.js'. Export your local IP address like so:

`module.exports.myLocalIp = xxx.xxx.xxx.xxx;`

[How to get local ip address?](http://www.wikihow.com/Find-Your-IP-Address-on-a-Mac)
## summary of changes
- Moved state out of the `app` component and into `MessageList`
- `SignIn` component:
  - had to patch up some things in order for me to start the app for Android
  - untabbed all the tabs =)
- Started a `MessageComposer` component, it's not being used anywhere
- Started a `MessageList` component. Depends on the message data also coming
  with user data
- Modified `MessageView` component. Also depends on the message data also
  coming with user data.
- Modified `Posts` component. Previously, this displayed a list of messages.
  Now, it wraps the `MessageList` component.
- `OAuthWebView` component:
  - had to patch up some things in order for me to start the app for Android
  - tabs to spaces :troll:

rafaelespinoza committed:
Convert Timeline/Post component to a class component

rafaelespinoza committed:
Patch up server routes

rafaelespinoza committed:
Move state.messages from App to Posts

rafaelespinoza committed:
Align Posts messages to top

rafaelespinoza committed:
Render the Posts component instead of Signin

Just doing this while I'm testing out the Posts component

rafaelespinoza committed:
Add a API helpers module

rafaelespinoza committed:
Get posts from DB to show up in timeline

rafaelespinoza committed:
Refactor API helper & its use in Posts component

rafaelespinoza committed:
Restore initial route

rafaelespinoza committed:
Move helpers/ to src/ & update references

rafaelespinoza committed:
Start work on MessageList component

rafaelespinoza committed:
Replace hard-coded ip address

rafaelespinoza committed:
Convert tabs to spaces

rafaelespinoza committed:
Begin moving state & messages out of Timeline/Posts into MessageList

rafaelespinoza committed:
Make things workable for Android dev environment

Unforutnately, React Native does not let you pass in environment
variables through the command line.

Am taking a cue from this StackOverflow answer:
http://stackoverflow.com/a/33134782/2908123

rafaelespinoza committed:
Remove console log statements from MessageList component

rafaelespinoza committed:
Update MessageList & Message component to display richer user data

rafaelespinoza committed:
Add a postMessage API helper method

rafaelespinoza committed:
Start work on MessageComposer element

rafaelespinoza committed:
Merge branch 'master' into feature-REBASED-Timeline

rafaelespinoza committed:
Merge branch 'master' into feature-components

rafaelespinoza committed:
Merge branch 'feature-REBASED-Timeline' into feature-components

pranaymohan commented on client/zestyreactnative/src/components/authentication/signin.js:
```diff
@@ -5,31 +5,33 @@ import {
   View
 } from 'react-native';
 import Button from '../common/button';
+var myLocalIp = require('../../helpers/scaffolding.js').myLocalIp;
```
I don't see a myLocalIp method or variable in scaffolding.js -- am I missing this, or is it gitignored? I only see border color method

pranaymohan commented on client/zestyreactnative/src/components/message/composer.js:
```diff
@@ -0,0 +1,45 @@
+      text: ''
+    };
+
+    this.displayText = this.displayText.bind(this);
```
no displayText method yet on this component, not sure if .bind will fail on an undefined function

pranaymohan commented on client/zestyreactnative/src/components/message/list.js:
```diff
@@ -0,0 +1,68 @@
+  }
+
+  render() {
+    var renderedMessage = ({ user, text, likes }, index) => {
```
I think this should be a component method called renderMessage(user, text, likes) that you invoke in the actual render function return statement. It's a bit too much going on in the actual render function right now.

This is not an immediate fix -- only fix after we get done with mvp :)

pranaymohan commented on client/zestyreactnative/src/components/message/list.js:
```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
```
good convention is to name the file after the component. So rename list.js -> messagelist.js

pranaymohan commented on client/zestyreactnative/src/components/authentication/signin.js:
```diff
@@ -5,31 +5,33 @@ import {
     this.onPress = this.onPress.bind(this);
   }
   render () {
-		return (
```
thank you for fixing my janky tabs 💯

rafaelespinoza committed:
Patch up some things

rafaelespinoza committed:
Rename component files as per their component class name

rafaelespinoza committed:
Merge branch 'master' into feature-components

Commit:
add background image and style sign in page

Commit:
check for accesstoken url, store token and redirect to posts view

rafaelespinoza committed:
Quick fix to Message

- Update the params to reflect the server response.
- Quick & dirty fix to handle any possible empty values.
- troubleshoot style warnings

rafaelespinoza committed:
Update props passed to Message from MessageList

rafaelespinoza committed:
Put borders around parts of individual message

rafaelespinoza committed:
Add navigator route to MessageComposer & a button from Posts

rafaelespinoza committed:
Start work on sending a message to server from MessageComposer

rafaelespinoza committed:
Make scaffolding borders thinner

rafaelespinoza committed:
Minor cleanups to Message components

rafaelespinoza committed:
Make Message layout a little more like Twitter

rafaelespinoza committed:
Let Button component take in styles from parent component

rafaelespinoza committed:
Clean up styles for Message component

rafaelespinoza committed:
Update Header & Button area of Posts component

NerdDiffer committed:
Send access token along with postMessage

NerdDiffer committed:
Use AsyncStorage to retrieve accessToken while creating Message

NerdDiffer committed:
Add hashtag png

NerdDiffer committed:
Work on styling the Signin page with Hashtag image

NerdDiffer committed:
Merge branch 'master' into feature-ConnectThings

NerdDiffer committed:
Merge remote-tracking branch 'upstream/master' into feature-components

NerdDiffer committed:
Merge branch 'master' into feature-components

NerdDiffer committed:
Merge remote-tracking branch 'upstream/master' into feature-ConnectThings

NerdDiffer committed:
Merge branch 'feature-ConnectThings' into feature-components

NerdDiffer committed:
Merge branch 'feature-StyleThings' into feature-components

NerdDiffer committed:
Merge branch 'master' into feature-components

pranaymohan commented on client/zestyreactnative/src/components/message/message.js:
```diff
@@ -0,0 +1,83 @@
+var Message = ({ User, text, likes }) => {
+  if (User) {
+    var { username, name, location, avatarUrl } = User;
+  } else {
```
Not asap, but need to figure out better solutions to default avatar, etc. For now this is great

pranaymohan commented on client/zestyreactnative/src/components/message/messageComposer.js:
```diff
@@ -0,0 +1,65 @@
+        var text = this.state.text;
+
+        postMessage(text, accessToken)
+          .then(message => this.props.navigator.pop())
```
need to keep an eye out for nested promises in the future (antipattern) -- but for now, this is good!

NerdDiffer committed:
Fix dark tint on MessageComposer view
